pkg/framework/proxies.go

```diff
@@ -0,0 +1,196 @@
+package framework
+
+import (
+	"context"
+	"errors"
+	configv1 "github.com/openshift/api/config/v1"
+	kappsapi "k8s.io/api/apps/v1"
```

Comment:
I notice this isn't imported as `appsv1` in this package, but it is in the rest of the codebases and this is the convention, would you mind updating to `appsv1`?